inference-engine/tests/functional/plugin/cpu/single_layer_tests/topk.cpp

```diff
@@ -0,0 +1,349 @@
+public:
+    static std::string getTestCaseName(testing::TestParamInfo<TopKLayerCPUTestParamsSet> obj) {
+        LayerTestsDefinitions::TopKParams basicParamsSet;
+        CPUSpecificParams cpuParams;
+        std::tie(basicParamsSet, cpuParams) = obj.param;
+
+        std::ostringstream result;
+        result << LayerTestsDefinitions::TopKLayerTest::getTestCaseName(
+                     testing::TestParamInfo<LayerTestsDefinitions::TopKParams>(basicParamsSet, 0));
+
+        result << CPUTestsBase::getTestCaseName(cpuParams);
+
+        return result.str();
+    }
+
+protected:
+    void SetUp() {
+        LayerTestsDefinitions::TopKParams basicParamsSet;
+        CPUSpecificParams cpuParams;
+        std::tie(basicParamsSet, cpuParams) = this->GetParam();
+
+        std::tie(inFmts, outFmts, priority, selectedType) = cpuParams;
+
+        InferenceEngine::SizeVector inputShape;
+        InferenceEngine::Precision netPrecision;
+        int64_t keepK, axis;
+        ngraph::opset4::TopK::Mode mode;
+        std::tie(keepK, axis, mode, sort, netPrecision, inPrc, outPrc, inLayout, inputShape, targetDevice) = basicParamsSet;
+
+        axis_idx = axis < 0 ? static_cast<size_t>(axis + static_cast<int64_t>(inputShape.size())) : static_cast<size_t>(axis);
+        top_k = static_cast<size_t>(keepK);
+        for (size_t i = 0; i < axis_idx; i++)
+            outer_size *= inputShape[i];
+        for (size_t i = axis_idx + 1; i < static_cast<size_t>(inputShape.size()); i++)
+            inner_size *= inputShape[i];
+
+        auto ngPrc = FuncTestUtils::PrecisionUtils::convertIE2nGraphPrc(netPrecision);
+        auto params = ngraph::builder::makeParams(ngPrc, {inputShape});
+        auto paramIn = ngraph::helpers::convert2OutputVector(
+                            ngraph::helpers::castOps2Nodes<ngraph::op::Parameter>(params));
+
+        auto k = std::make_shared<ngraph::opset3::Constant>(ngraph::element::Type_t::i64, ngraph::Shape{}, &keepK);
+        auto topk = std::dynamic_pointer_cast<ngraph::opset4::TopK>(
+                std::make_shared<ngraph::opset4::TopK>(paramIn[0], k, axis, mode, sort));
+        topk->get_rt_info() = getCPUInfo();
+
+        ngraph::ResultVector results;
+        for (int i = 0; i < topk->get_output_size(); i++) {
+            results.push_back(std::make_shared<ngraph::opset4::Result>(topk->output(i)));
+        }
+        function = std::make_shared<ngraph::Function>(results, params, "TopK");
+    }
+
+    void Validate() {
+        auto expectedOutputs = CalculateRefs();
+        const auto &actualOutputs = GetOutputs();
+
+        if (expectedOutputs.empty()) {
+            return;
+        }
+
+        IE_ASSERT(actualOutputs.size() == expectedOutputs.size())
+        << "nGraph interpreter has " << expectedOutputs.size() << " outputs, while IE " << actualOutputs.size();
+
+        // Spec TopK_3.md allows to use unstable sorting, thus
```

Comment:
As far as I understand, the default Compare and Validate methods are not used here (and in other updated tests) because of unstable sorting, so we need to sort before comparison.
I can imagine that the similar logic will be needed for other plugins as well, so maybe instead we could extend  Compare methods in layer_test_utils so we could apply some function (sort in this case) to compared tensors (expected and actual) before they are compared.
Otherwise this leads to too many code duplication.
Also just wondering, in general, can this unstable sorting result in accuracy deviations?
This is just a point for discussion, I am not asking you to do this right away.

Reply:
Good point!
1. About extend Compare methods.  Fixed.
2. About unstable sorting. I think stable sorting is necessary when dealing with multiple keys. Since our topk node is about single key sorting, I don't think there will be accuracy deviations applying unstable sorting. Though from the perspective of validation, with unique expected output, stable sorting is more welcomed. Yet there is always tradeoff between stablity and effiency. Here we consider more from the perspective of performance to choose unstable sorting.
Further discussion is always welcomed. Thanks Egor!

Reply:
Thanks you for explanation.
Still would be good if @dmitry-gorokhov provide his opinion on this.
As for me, I think we copy paste to much code just to slightly (even it is major difference) change the behavior.